addons/website_slides/models/slide_channel.py

```diff
@@ -70,7 +70,7 @@ def _recompute_completion(self):
                 continue
             elif not record.completed and record.completed_slides_count >= record.channel_id.total_slides:
                 completed_records += record
-            elif record.completed and record.completed_slides_count < record.channel_id.total_slides:
+            elif self.env.context.get("slides_marked_uncompleted") and record.completed and record.completed_slides_count < record.channel_id.total_slides:
                 uncompleted_records += record
 
```

Comment:
Hello @nboulif , If I understand it correctly, the idea is to update the completion of attendees having completed the course **only if** they manually mark slides uncompleted by themselves, right? in that case, the records will be in uncompleted_records, but as their completion is set above, will it not remain 100% ? I think your test does not cover the case:

Was completed -> mark one slide uncompleted -> check completion is not 100 %

Or maybe is there a trigger when 'completed' changes that I am missing?

Thanks for the info and cheers! :)

Reply:
Hello @Odoonan , thanks for the check!
Indeed, missed this case. The issue is coming from the fact that the 'slide.slide.partner' has already his completed set to False but not the `'slide.channel.partner'. I updated the fix and added the step in the test :+1:

Reply:
Thanks for the update @nboulif 

From 16.3 on,  as I remember we chose to never update the 'completed' status once the course is completed. As discussed live with @flch-odoo , It may be worth double checking with the product owner (FRGI) if we want to consider that part [set channel uncomplete when manually marking a slide uncompleted] as a fix, or if we drop that part.  He is off today, though ^^

Cheers!